src/regex_dna.rs

```diff
@@ -7,33 +7,35 @@
 
 extern crate regex;
 
+use regex::bytes::Regex;
+
 use std::io::{self, Read};
 use std::sync::Arc;
 use std::thread;
 
-macro_rules! regex { ($re:expr) => { ::regex::Regex::new($re).unwrap() } }
+macro_rules! regex { ($re:expr) => { Regex::new($re).unwrap() } }
 
 fn main() {
-    let mut seq = String::with_capacity(50 * (1 << 20));
-    io::stdin().read_to_string(&mut seq).unwrap();
+    let mut seq = Vec::with_capacity(50 * (1 << 20));
+    io::stdin().read_to_end(&mut seq).unwrap();
     let ilen = seq.len();
 
-    seq = regex!(">[^\n]*\n|\n").replace_all(&seq, "");
+    seq = regex!(r">[^\n]*\n|\n").replace_all(&seq, &b""[..]);
```

Comment:
Nope. Byte string literals are not user friendly because their type is a fixed size array instead of a slice.

One possible fix here is for the `regex` crate to add a whole bunch of impls for `[u8; N]`, but I didn't want to go down that path.